src/test/java/com/google/sps/data/ContestTest.java

```diff
@@ -0,0 +1,157 @@
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void oneContest_twoCandidates_fromJSONObject() {
+    JSONObject contestJSON =
+        new JSONObject(
+            "{\""
+                + Contest.TYPE_JSON_KEYWORD
+                + "\":\"General\",\""
+                + Contest.NAME_JSON_KEYWORD
+                + "\":\"Governer\",\""
+                + Contest.CANDIDATES_JSON_KEYWORD
+                + "\":["
+                + "{\"name\": \"name1\",\"party\": \"party1\",\"candidateUrl\": \"www.siteOne.com\"},"
+                + "{\"name\": \"name2\",\"party\": \"party2\",\"candidateUrl\": \"www.siteTwo.com\"}"
+                + "]}");
+
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    Contest newContest = Contest.fromJSONObject(ds, contestJSON);
+    Set<Long> candidateIds = newContest.getCandidates();
+    Assert.assertEquals(candidateIds.size(), 2);
+    Assert.assertEquals(newContest.getName(), "Governer");
+    Assert.assertEquals(newContest.getDescription(), "");
+  }
+
+  @Test
+  public void oneContest_twoCandidates_JsonString() {
+    Candidate candidateOne =
+        Candidate.builder()
+            .setName("Jane Doe")
+            .setPartyAffiliation("Green")
+            .setPlatformDescription("")
+            .setCampaignSite("www.janedoe.org")
+            .build();
+
+    Candidate candidateTwo =
+        Candidate.builder()
+            .setName("James Doe")
+            .setPartyAffiliation("Tea")
+            .setPlatformDescription("")
+            .setCampaignSite("www.jamesdoe.org")
+            .build();
+
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    long candidateOneId = candidateOne.addToDatastore(ds);
+    long candidateTwoId = candidateTwo.addToDatastore(ds);
+
+    ImmutableSet<Long> candidateIdSet =
+        ImmutableSet.<Long>builder().add(candidateOneId).add(candidateTwoId).build();
+
+    Contest newContest =
+        Contest.builder()
+            .setName("Governer")
+            .setCandidates(candidateIdSet)
+            .setDescription("Race for California Governer")
+            .build();
+
+    JsonElement returnedContestJSON = JsonParser.parseString(newContest.toJsonString(ds));
+
+    JsonElement expectedContestJSON =
+        JsonParser.parseString(
+            "{\""
+                + Contest.NAME_ENTITY_KEYWORD
+                + "\":\"Governer\", \""
+                + Contest.CANDIDATES_ENTITY_KEYWORD
+                + "\":["
+                + candidateOne.toJsonString()
+                + ","
+                + candidateTwo.toJsonString()
+                + "],\""
+                + Contest.DESCRIPTION_ENTITY_KEYWORD
+                + "\":\"Race for California Governer\"}");
+
+    Assert.assertEquals(expectedContestJSON, returnedContestJSON);
+  }
+
+  @Test
+  public void testFromEntity() {
+    ImmutableSet<Long> idSet = ImmutableSet.<Long>builder().add(1L).add(2L).build();
```

Comment:
I think you can just do:

ImmutableSet.of(1L, 2L);

for small numbers of elements.

Reply:
done